### PR TITLE
feat(plasma-core/plasma-ui): processing price periodicity

### PR DESCRIPTION
**Pull request by Yakutoc**
Добавить возможность указывать в цене регулярность платежа.

Возможные значения:

```
...
periodicity?: '/ день' | '/ неделя' | '/ месяц' | '/ год' | ReactElement;
```

так же есть возможность прокинуть свой компонент для отображения регулярности платежа.

:baby_chick: Download canary assets:

[colors--canary.220.3488417376.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.220.3488417376.xml)  
[color_sbermarket_ios-swift--canary.220.3488417376.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.220.3488417376.swift)  
[color_sbermarket_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.220.3488417376.kt)  
[color_sbermarket_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.220.3488417376.ts)  
[color_sberprime_ios-swift--canary.220.3488417376.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.220.3488417376.swift)  
[color_sberprime_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.220.3488417376.kt)  
[color_sberprime_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.220.3488417376.ts)  
[PlasmaTokensColor--canary.220.3488417376.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.220.3488417376.swift)  
[typo_mage_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.220.3488417376.kt)  
[typo_mage_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.220.3488417376.ts)  
[typo_plasma_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.220.3488417376.kt)  
[typo_plasma_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.220.3488417376.ts)  
[typo_ruler_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.220.3488417376.kt)  
[typo_ruler_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.220.3488417376.ts)  
[typo_sage_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.220.3488417376.kt)  
[typo_sage_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.220.3488417376.ts)  
[typo_sbermarket_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.220.3488417376.kt)  
[typo_sbermarket_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.220.3488417376.ts)  
[typo_soulmate_kotlin--canary.220.3488417376.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.220.3488417376.kt)  
[typo_soulmate_react-native--canary.220.3488417376.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.220.3488417376.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.114.0-canary.220.3488417376.0
  npm install @salutejs/plasma-core@1.82.0-canary.220.3488417376.0
  npm install @salutejs/plasma-icons@1.108.0-canary.220.3488417376.0
  npm install @salutejs/plasma-temple@1.112.0-canary.220.3488417376.0
  npm install @salutejs/plasma-ui@1.145.0-canary.220.3488417376.0
  npm install @salutejs/plasma-web@1.142.0-canary.220.3488417376.0
  npm install @salutejs/plasma-sb-utils@0.80.0-canary.220.3488417376.0
  # or 
  yarn add @salutejs/plasma-b2c@1.114.0-canary.220.3488417376.0
  yarn add @salutejs/plasma-core@1.82.0-canary.220.3488417376.0
  yarn add @salutejs/plasma-icons@1.108.0-canary.220.3488417376.0
  yarn add @salutejs/plasma-temple@1.112.0-canary.220.3488417376.0
  yarn add @salutejs/plasma-ui@1.145.0-canary.220.3488417376.0
  yarn add @salutejs/plasma-web@1.142.0-canary.220.3488417376.0
  yarn add @salutejs/plasma-sb-utils@0.80.0-canary.220.3488417376.0
  ```
</details>

### CODE REVIEW

**Comment by Yakutoc** [comment]
После того как это PR вольем.

Мне останется всего лишь подлить мастер в свою ветку где уже реализована поддержка **Periodicity** для компонента **ProductCart**.

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/

File: packages/plasma-core/src/components/Price/Price.component-test.tsx [hunk 1]
```diff
@@ -69,4 +75,26 @@ describe('plasma-core: Price', () => {
 
         cy.matchImageSnapshot();
     });
+
+    it('processing periodicity', () => {
+        mount(
+            <CypressTestDecorator>
+                <style>{`
+                    body {
+                        font-family: "SB Sans Text", sans-serif;
```
Comment (neretin-trike) [inline packages/plasma-core/src/components/Price/Price.component-test.tsx:84]: а для чего здесь добавлять font-family?

File: packages/plasma-core/src/components/Price/Price.tsx [hunk 2]
```diff
@@ -43,9 +21,10 @@ export const Price: React.FC<PriceProps> = ({
     currency = 'rub',
     minimumFractionDigits = 0,
     stroke = false,
+    periodicity,
     ...rest
 }) => (
     <StyledPrice $stroke={stroke} {...rest}>
-        {formatCurrency(children, currency, minimumFractionDigits)}
+        {formatCurrency(children, currency, minimumFractionDigits)} {periodicity || null}
```
Comment (neretin-trike) [inline packages/plasma-core/src/components/Price/Price.tsx:28]: *минор*

а почему не просто {periodicity}?

Comment (Yakutoc) [reply packages/plasma-core/src/components/Price/Price.tsx:28]: Тоже верно. Сначала сделал как защита от дурака, а сейчас думаю что можно и проще )

Спасибо.

File: packages/plasma-core/src/components/Price/Price.component-test.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-core/src/components/Price/Price.component-test.tsx:84]: @neretin-trike Тут сделал по аналогии с остальными тестами.  Суть проблемы со стилями не изучал еще.

Comment (neretin-trike) [reply packages/plasma-core/src/components/Price/Price.component-test.tsx:84]: а, окей 👌

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-core/src/components/Price/Price.component-test.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/Price/Price.component-test.tsx:84]: а завели задачку

File: packages/plasma-core/src/components/Price/Price.component-test.tsx [hunk 3]
```diff
@@ -1,11 +1,17 @@
 import React from 'react';
 import { mount, CypressTestDecorator, getComponent, PadMe } from '@salutejs/plasma-cy-utils';
 
+enum Periodicity {
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Price/Price.component-test.tsx:4]: а почему его не импортировать из самого компонента?

File: packages/plasma-core/src/components/Price/types.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    stroke?: boolean;
+    /**
+     * Минимальное количество цифр после десятичного разделителя.
+     */
+    minimumFractionDigits?: number;
+    /**
+     * Указывает регулярность платежа, идет после цены.
+     * Например - 990Р / день.
+     */
+    periodicity?: Periodicity | ReactElement;
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Price/types.ts:33]: чет печально это конечно в доке выглядит 
Screenshot 2022-11-17 at 10 24 26

File: packages/plasma-ui/src/components/Price/Price.stories.tsx [hunk 5]
```diff
@@ -10,6 +11,10 @@ const currencyOptions = ['rub', 'usd', 'eur'];
 
 const propsToDisable = ['children', 'theme', 'as', 'forwardedAs'];
 
+enum PeriodicityDefault {
+    default = '',
```
Comment (Yeti-or) [inline packages/plasma-ui/src/components/Price/Price.stories.tsx:15]: а это типа не заданное значение?

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-220/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-220/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/temple-storybook/

  </details>

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-220/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-220/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/temple-storybook/

  </details>

**Review by Yeti-or — CHANGES_REQUESTED** [review]
Screenshot 2022-11-17 at 14 26 52

Это не выглядит как плавающий тест

**Comment by Yakutoc** [comment]
> Screenshot 2022-11-17 at 14 26 52
> 
> Это не выглядит как плавающий тест
Уже разобрались в чем проблема.

**Comment by Yeti-or** [comment]
@Yakutoc и в чем ?

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-220/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-220/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/temple-storybook/

  </details>

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-220/

  </details>

File: packages/plasma-core/src/components/Price/Price.component-test.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/Price/Price.component-test.tsx:77]: TODO: github issue плиз

File: packages/plasma-ui/src/components/Price/Price.stories.tsx [hunk 6]
```diff
@@ -21,17 +21,40 @@ export default {
                 options: currencyOptions,
             },
         },
+        periodicity: {
+            control: {
+                type: 'select',
+                options: ['/ день', '/ неделя', '/ месяц', '/ год', ''],
```
Comment (Yeti-or) [inline packages/plasma-ui/src/components/Price/Price.stories.tsx:27]: пустая строка выглядит как хак конечно, я правильно понял что оно не умеет быть не выбраным?

File: packages/plasma-core/src/components/Price/types.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    stroke?: boolean;
+    /**
+     * Минимальное количество цифр после десятичного разделителя.
+     */
+    minimumFractionDigits?: number;
+    /**
+     * Указывает регулярность платежа, идет после цены.
+     * Например - 990Р / день.
+     */
+    periodicity?: '/ день' | '/ неделя' | '/ месяц' | '/ год' | React.ReactElement;
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Price/types.ts:26]: Screenshot 2022-11-17 at 15 24 01
вот же 💩

Comment (Yeti-or) [reply packages/plasma-core/src/components/Price/types.ts:26]: ладно видимо забьем на это

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-core/src/components/Price/types.ts (hunk 7 above)
Comment (Yakutoc) [reply packages/plasma-core/src/components/Price/types.ts:26]: @Yeti-or Давай пока так. И я поищу способ как это упростить.

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-220/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/temple-storybook/

  </details>

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-220/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-220/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-220/
  ui storybook: http://plasma.sberdevices.ru/pr-220/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-220/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-220/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-220/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-220/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-220/temple-storybook/

  </details>

**Comment by Yeti-or** [comment]
:shipit:

**Pull request merged**